<launch>
<arg name="robot"/>
<!-- autostart: true to start the PPLP at startup -->
<arg name="autostart" default="false"/>
<!-- eventgui: true to create a GUI that allows to start the PPLP -->
<arg name="eventgui"  default="false"/>
<!-- marker: true to create a node converting the published PPL into a Vizualisation marker -->
<arg name="marker"    default="false"/>
<arg name="viewer"    default="false"/>
<arg name="rgb_topic" default="rgb"/>

<group ns="$(arg robot)">
  <!-- 2D face PPLP -->
  <node pkg="people_detection_vision" type="face_detector2d_pplp.exe"
    name="face_detector2d_pplp"  output="screen"
    args="_rgb_topic:=$(arg rgb_topic)
          _display:=false" />

  <!-- autostart: publish and latch starting event for skill, then die -->
  <node if="$(arg autostart)" pkg="rostopic" type="rostopic" name="face_detector2d_pplp_starter"
        output="screen" args="pub FACE_DETECTOR2D_PPLP_START std_msgs/Int16 '{data: 0}' --latch  -1"/>

  <!-- conversion to ROS rviz marker -->
  <node if="$(arg marker)" pkg="people_utils" type="ppl2marker.exe"
        output="screen" name="ppl2marker_face2d_pplp"
        args="_red:=1 _green:=0 _blue:=0 _input_topic:=face_detector2d_pplp/ppl" />

  <!-- event GUI -->
  <node if="$(arg eventgui)" pkg="ad_core" type="launcher_qt_event_sender_interface.exe"
        name="face_detector2d_pplp_event_gui"
        output="screen"
        args='event_keys="|FACE_DETECTOR2D_PPLP_START;FACE_DETECTOR2D_PPLP_START;0
                          |FACE_DETECTOR2D_PPLP_STOP;FACE_DETECTOR2D_PPLP_STOP;0"'/>

  <!-- PPL viewer -->
  <node if="$(arg viewer)" pkg="people_utils" type="ppl_viewer.exe"
    name="ppl_viewer"  output="screen"
    args="_ppl_topics:='face_detector2d_pplp/ppl'
          _reprojection_mode:=xz
          _width:=480 _height:=800
          _xmin:=-3 _xmax:=3 _ymin:=-1 _ymax:=8" />
</group>
</launch>
